app/src/components/EmptyList.tsx

```diff
@@ -1,42 +1,41 @@
-import { useTheme, testIdWithKey, Button, ButtonType, useStore } from 'aries-bifold'
-import React, { useCallback } from 'react'
+import React, { useCallback, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { Text, View } from 'react-native'
+import { Text, View, DeviceEventEmitter } from 'react-native'
 
 import EmptyWallet from '../assets/img/emptyWallet.svg'
-import { BCState, BCDispatchAction } from '../store'
+import { useTheme, testIdWithKey, Button, ButtonType } from 'aries-bifold'
+import { BCWalletEventTypes } from '../events/eventTypes'
 export interface EmptyListProps {
   message?: string
 }
 
 const EmptyList: React.FC<EmptyListProps> = ({ message }) => {
-  const { t } = useTranslation()
-  const { ListItems } = useTheme()
-  const [store, dispatch] = useStore<BCState>()
+    const { t } = useTranslation()
+    const { ListItems, } = useTheme()
+    const [addCredentialPressed, setAddCredentialPressed] = useState<boolean>(false)
 
-  const addCredentialPress = useCallback(() => {
-    dispatch({
-      type: BCDispatchAction.ADD_CREDENTIAL_PRESSED,
-      payload: [true],
-    })
-  }, [])
+    DeviceEventEmitter.addListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED,
```

Comment:
I **think** will this lead to multiple subscribers? Wondering if we need to remove the listener if it exists before adding it unless we know it has been cleaned up. My though is something like this.

```TypeScript
useEffect(() -> {

DeviceEventEmitter.addListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED,
        (value?: boolean) => {
            const newVal = value === undefined ? !addCredentialPressed : value
            setAddCredentialPressed(newVal)
        }
    )

    return {
	DeviceEventEmitter.removeListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED)
     }
}
```

Reply:
@bryce-mcmath You know react - thoughts?

Reply:
agree, should unsubscribe in a cleanup function (assuming the above is just pseudocode, I don't think it's actually valid. You would need to add an empty dep array and return an actual function from the useEffect callback)

Reply:
@jleach I'm not quite sure what you mean, the add listener is in the root of the component, not inside a useEffect or anything. It will only get called once

Reply:
nevermind, you guys are right. I'll update